### DISCUSSION TITLE
Support for choice option in Integer Field too

### DISCUSSION THREAD

**Discussion by sumitsharansatsangi** [piccolo-orm/piccolo#871] [Ideas]
Piccolo currently supports choice option only for `Varchar` Column type , It will be good , if it can support for Integer Type too.

**Comment by sinisaos** [comment]
@sumitsharansatsangi Sorry if I don't understand well, but you can already use choices in `Integer` column. [Example](https://github.com/piccolo-orm/piccolo_admin/blob/ecfd52de2aa3dec2b00d4e0f4f059a9d07470cd2/piccolo_admin/example.py#L208) from Piccolo Admin.

**Comment by sumitsharansatsangi** [comment]
Piccolo Supports Choice for IntegerField. It was my mistake. I have not checked it properly.

**Reply by dantownsend** [reply]
If you have this:

```python
class MyTable(Table):
    class Choices(int, Enum):
        choice_1 = 1
        choice_2 = 2

    my_column = Integer(choices=Choices)
```

In Piccolo Admin the options are shown as `choice 1`, `choice 2` etc (i.e. the enum keys), and not the values (`1`, `2`) which is what's stored in the database.

If you wanted the choices to be numbers from 1 to 3 in Piccolo Admin, and show them as '1', '2', '3' ... you should use this:

https://github.com/piccolo-orm/piccolo/blob/master/piccolo/columns/choices.py

I'm not sure if it's in the docs, but it will let you do this:

```python
from piccolo.columns.choices import Choice

class MyTable(Table):
    class Choices(Enum):
        choice_1 = Choice(1, '1')
        choice_2 = Choice(2, '2')
        choice_3 = Choice(3, '3')

    my_column = Integer(choices=Choices)
```

**Reply by sumitsharansatsangi** [reply]
@dantownsend , Thanks for suggesting the second method, I was not aware of this, and it's not in the docs.  I think the second method has more use cases than the first one. Please add it in the docs.
